app/src/main/java/org/opencord/aaa/impl/AaaManager.java

```diff
@@ -264,6 +280,29 @@ private void configurePacketCustomizer() {
                 break;
         }
     }
+    private boolean isValidValidator(RADIUS radiusPacket) {
+        log.info("radiusPacket.getAttribute(RADIUSAttribute.RADIUS_ATTR_MESSAGE_AUTH)---" +
+                radiusPacket.getAttribute(RADIUSAttribute.RADIUS_ATTR_MESSAGE_AUTH).getValue());
+        return radiusPacket.checkMessageAuthenticator(AaaManager.this.radiusSecret);
+        }
+    public void checkForPacketFromUnknownServer(String hostAddress) {
+        try {
+            InetAddress address;
+            if (newCfg.radiusHostName() != null) {
+                 address = InetAddress.getByName(newCfg.radiusHostName());
```

Comment:
All this calculation of getting configured AAA server address should not be done on receipt of each Radius packet. Can just be done once. So, define a instance variable String configuredAaaServerAddress in this class and populate in activate method. Then, only the comparison should be done in this method